cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -237,6 +238,24 @@ def _set_default_instance_role(self):
             ],
         }
 
+        if build.components:
+            for custom_component in build.components:
+                # Check custom component is script, and the url is S3 url
+                if custom_component.type == "script" and custom_component.value.startswith("s3"):
+                    bucket_info = get_bucket_object_name(custom_component.value)
```

Comment:
`get_bucket_object_name` -> `parse_bucke_url` ? we are not getting just the object name

Reply:
done